lib/devise-security/models/secure_validatable.rb

```diff
@@ -82,6 +82,10 @@ def self.assert_secure_validations_api!(base)
         raise "Could not use SecureValidatable on #{base}" unless base.respond_to?(:validates)
       end
 
+      def self.required_fields(_klass)
+        []
```

Comment:
Seems like this should use `klass` to conditionally require based on the validations starting on line 25?

Reply:
Example from Devise: https://github.com/heartcombo/devise/blob/main/lib/devise/models/confirmable.rb#L72

Reply:
Thank you so much for the review @dillonwelch ❤️ 

I think that `self.required_fields` should only include the attributes that are absent in the devise model table by default and required by this module. Since the `email` and `password` fields are added by the default devise migration,  we should be okay to keep this empty. It's empty for the `Devise::Models::Validatable` as well https://github.com/heartcombo/devise/blob/e2242a95f3bb2e68ec0e9a064238ff7af6429545/lib/devise/models/validatable.rb#L22.
Please let me know if I'm missing anything @dillonwelch. 🙏

Reply:
Also, the tests for the ruby `head` version i.e. `3.4.0` are failing because of the dependency of `minitest` on `mutex_m` which has been moved from default gems to bundled gems. Should I fix it in this PR, I'm thinking about adding the following to `gemspec` file.
```ruby
s.add_development_dependency 'mutex_m'
```

Reply:
Required fields: ok, that makes sense

Gem issue: Yes please!!!!!